pkg/agent/core/ngt/service/ngt.go

```diff
@@ -135,9 +135,19 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 		}
 	}
 
-	n.kvs = kvs.New()
 	n.dim = cfg.Dimension
 
+	n.kvs = kvs.New(func() []kvs.Option {
+		if cfg.KVSDB.Concurrency < 1 {
+			return []kvs.Option{
+				kvs.WithErrGroup(n.eg),
+			}
+		}
+		return  []kvs.Option{
+			kvs.WithConcurrency(cfg.KVSDB.Concurrency),
```

Comment:
I don't know but it should be like:
```suggestion
			kvs.WithErrGroup(n.eg),
			kvs.WithConcurrency(cfg.KVSDB.Concurrency),
```

Reply:
Good question, n.eg usually contains a global errgroup, so when you put it in kvsdb and enable Limitation, it will have a global limit, so if Concurrency < 1 (Unlimited), you can use global errgroup. However, in case of limited (Concurrency is set),  errgroup is created in kvsdb and Limitation is performed.
So, we don't need to set errgroup here.

Reply:
ok, Thanks!
Please write it as a comment in the code.

Reply:
got it.